components/tools/OmeroWeb/omeroweb/webgateway/views.py

```diff
@@ -1212,31 +1223,46 @@ def wrap(request, *args, **kwargs):
         try:
             server_id = kwargs.get('server_id', None)
             if server_id is None:
-                server_id = request.session['connector'].server_id
-            kwargs['server_id'] = server_id
+                if 'connector' in request.session:
```

Comment:
I've created a new @json_response decorator for all /api methods so I don't need this change above now. Reverted.